kubernetes/kubernetes_service.go

```diff
@@ -343,3 +379,17 @@ func (in *IstioClient) getDeployments(namespace string, deploymentsChan chan dep
 	deployments, err := in.k8s.AppsV1beta1().Deployments(namespace).List(emptyListOptions)
 	deploymentsChan <- deploymentsResponse{deployments: deployments, err: err}
 }
+
+func (in *IstioClient) getPods(namespace string, selector map[string]string, podsChan chan podsResponse) {
+	selectorQuery := labels.Set(selector).String()
+	pods, err := in.GetPods(namespace, selectorQuery)
+	podsChan <- podsResponse{pods: pods, err: err}
+}
+
+func (in *IstioClient) getServicesByApp(namespace string, selector map[string]string, serviceChan chan servicesResponse) {
+	cfg := config.Get()
+	serviceSelectorName := labels.Set(selector).Get(cfg.AppLabelName)
+	serviceSelector := labels.Set{cfg.AppLabelName: serviceSelectorName}
```

Comment:
Two remarks here:
- `serviceSelectorName := labels.Set(selector).Get(cfg.AppLabelName)` cuold be written shorter I guess: `serviceSelectorName, ok := selector[cfg.AppLabelName]`
- If there is no app label (so, the `ok` above is false) I guess we should send `nil` on the channel?